### ISSUE TITLE
Street Fighter II Custom OST WIP!:)

### ISSUE THREAD

**Issue by KMFDManic** [KMFDManic/mame2003-xtreme#16]
Hello, @grant2258 and @arcadez2003 and @gpstar81 !  You ready for the biggest, baddest, best addition to MAME 2003 Plus/Xtreme, to date:)  How about Custom OST Support for Street Fighter II, the venerable Arcade Classic...that I know we all played hundreds of hours on, be it Arcade or Home Ports!  Big Blue Frontend worked this in, and was nice enough to try to get things fixed up so we can have a better chance of this working out well on the Mini Classics!  Example Test Video of it in action:

https://www.youtube.com/watch?v=134slPc0Sug

I would like to do a few variants, so people can swap out the OSTs, as I would like to have THIS song for Ken's Stage!

https://www.youtube.com/watch?v=v9tvzw3E5tw&t=25s

gpstar81 also suggested I use a music remix of Mighty Wings, as an alternative that people might enjoy more.  I will do it both ways!

I will test these changes on the Mini Classics, whereas, you guys, if interested, can help verify they work on PC!    Once you confirm you are interested, I will give temporary link to the fixed up source files, so you can grab them.  Then, once everything is said and done and cleaned up, will commit them to the normal source commits!  

Kyland:)

**Comment by ghost** [comment]
I have a fork of xtreme on my repo so no problems testing it after youve done your rounds of testing on the pc at all. happy new year to you all!

**Comment by KMFDManic** [comment]
@grant2258 Will temporarily post the files here, and you and/or @arcadez2003 can confirm you've grabbed them.  Want them to be properly implemented before committing officially.  Note, the "Additional Refactoring Files" are meant to help with dual usage of ffight.zip and sf2.zip

https://we.tl/t-Lzvdqu2tsK

**Comment by ghost** [comment]
you want this tested on extrem or plus ?

**Comment by KMFDManic** [comment]
@grant2258 I would actually like this to be implemented for both.  Gpstar is helping, as well.  We should be able to find a median to have a solid set-up for SFII between the two Cores.  I think this will be incredibly fun to test and play and showcase:)

**Comment by ghost** [comment]
im not sure which source file i should be using here there are three can you give me some background into it i can just test if each sorce file is working if you prefer that

**Comment by KMFDManic** [comment]
I didn't get a good chance to dig too deep into them yet, @grant2258 Had been working additional hours due to the Holidays.  Let me know where you stand, and we will continue from there.  You did a friggin' fantastic job helping clear up the Samples and other issues in the past.  MAME 2003 Xtreme/Plus will always have your "quality seal" on them, period.

**Comment by KMFDManic** [comment]
@grant2258 @gpstar81 went into overdrive, and was able to convert from c++ to mame c code, and this should work out better for us on MAME 2003 Plus/Xtreme!

https://github.com/gpstar81/mame2003-xtreme/commit/92c6daf89bf3a9e6daf0116416488a8bd7b85811

I am gonna test things now.  Let me know how things turn out on your end, then we can try to work this into Plus/Xtreme for realz:)

I am very much hoping our other recent additions do not conflict with any of this, of course!  But, here we have, the coolest, biggest, baddest Custom OST, yet!:)

**Comment by ghost** [comment]
One note i will say im not sure who implemented the sample_stop(a); loop but we are only using two channels sample_stop(0); and sample_stop(1); will stop the samples

**Comment by gpstar81** [comment]
you are right. it's been changed on my repo to sample_stop(0) and sample_stop(1).

**Comment by KMFDManic** [comment]
@grant2258 this is exactly why I absolutely value your final say in this.  You and arcadez2003 and markwkidd have all been amazing with me in the past.  This WIP will be very nice once you add your TLC to it:)

**Comment by ghost** [comment]
i do appoligize for the stopall the samples where merging your way is a lot better! great work you two guys as usual!

**Comment by ghost** [comment]
Ok little update first of all the volume is so much more pleasing. The normal sounds are also playing samples arent there. The slow sound is cutting out sometimes. Usually starts on the first fight on round two sometimes it comes back on or it just stays off until the fast music comes on

**Comment by gpstar81** [comment]
What system are you testing on? PC? ARM build for a SNES classic or equivalent ?

**Comment by ghost** [comment]
can do all 3 linux windows and arm I was testing on windows

**Comment by gpstar81** [comment]
What happens if you build from my repo and use the module from that?

I couldn’t get KMFDManics repo to build on my system, and I didn’t want to spend the time debugging the issue so I worked off my old source from a year ago. His was giving me the following problems when linking:

Undefined symbols for architecture x86_64:
  "_osd_get_mastervolume", referenced from:
      _onscrd_volume in usrintrf.o
  "_osd_set_mastervolume", referenced from:
      _onscrd_volume in usrintrf.o
  "_osd_sound_enable", referenced from:
      _mame_pause in mame.o
      _handle_user_interface in usrintrf.o
ld: symbol(s) not found for architecture x86_64

**Comment by ghost** [comment]
I havent tried builing his repo will give it a try now 

do this in his repo
git fetch origin
git reset --hard origin/master

make clean 
make -j8

edit his repo builds fine for me

**Comment by gpstar81** [comment]
Same problem again with his. I'm building on a mac btw.

I'll try building yours.

**Comment by gpstar81** [comment]
Your's built fine on my system.

**Comment by ghost** [comment]
i already have your commits in

**Comment by gpstar81** [comment]
Ah anyways, I ran yours, no issues with the samples on my end.

**Comment by ghost** [comment]
play a few rounds it took a while for me the first time try win really quick it happened with ehonda  and zangief blanka as well. Ill try a clean compile tomorrow cause i was already compiled just recompiled the driver mame can be funny if you dont do a clean build at times. Will update you when i do this

**Comment by gpstar81** [comment]
I’ve done several play throughs no problem.

Which rom are you using? sf2 or sf2ce? Can you do a video when the problem happens?

**Comment by KMFDManic** [comment]
@gpstar81 @grant2258 Funny bug, if you have infinite health on, in my video.  Not worried about it, though!  Game runs great, will get more recent commits added.  But, here is the end result!  Thanks, guys!  And, thanks Big Blue Frontend, too!

https://www.youtube.com/watch?v=ooURAXmKY60

**Comment by ghost** [comment]
ive done a test video where this happens its happening on windows and linux for me this was recorded in linux dont know how to record it on windows im a video recorder n00b. ill add a link when its uploaded

https://drive.google.com/open?id=1LlOdWaOeUP_Vmomn4ENYdOmV7euqd-XZ

need to get some sleep for worked glanced at the code 
maybe add a condition for

else if(sample_playing(0) == 0 && sample_playing(1) == 0 && fadingMusic ) fadingMusic  = false;

edit:

traced this a bit the fading 

https://github.com/grant2258/mame2003-xtreme/blob/26f432554bc580d1b0584e660cddec51edafebd1/src/drivers/cps1.c#L975-L976

commenting this out just makes a small gap then the music starts maybe intentional or the game timing is off?

anyway to get this to reproduce i start the game put three credits in choose ehonda first fight is chungli then blanks and it tiggers

**Comment by KMFDManic** [comment]
@grant2258 @gpstar81 Try these "better" samples, which are complete and have FM Towns, and crucial 3DO tracks.  Thanks, both of you, as well as Big Blue Frontend, for doing a great job with the original c++ coding and samples and hooks!

https://archive.org/download/bjjvh/sf2.zip

Unless anything is changing, I will get things committed for the Next Release:)

**Comment by arcadez2003** [comment]
@KMFDManic @grant2258

Hey fellas hope ya had a good new year, could ya's do me a quick favour.?? just boot Dig Dug
and lemmie know if it works for ya's, as a dev i know called Madmab has had trouble getting it 
to work on the genesis mini when using the MAME2003+ core.

He's gonna try it on the SNES Classic next so i thought it might be good to hear back from
more than one person whether it's good or not.

Cheers!!

**Comment by ghost** [comment]
ill compile the latest for pi havent used it in a while so will be an older version i have installed for arm. It works fine in windows. I tested the digdug set.

edit working fine on the pi as well mate.

**Comment by arcadez2003** [comment]
Thanks for trying it out cheers, might be one of those platform specific issues then maybe.

**Comment by ghost** [comment]
@arcadez2003  one little thing to keep in mind it could be his rom is bad  the set was changed in the namco80s update we done. I get a feeling that could be the issue shoot him the plus rom to test with. 

edit notice you done a gng 4way update is a lot better this way i changed my sanwa to 4way and i found the game a lot easier even playing with the 4way emulation on helped it leaps n bounds

**Comment by arcadez2003** [comment]
Aye i told him as much but he was using the xbox MAME rom for Dig Dug which i knew would
have been correct for the Namco update, anyway he's got it working now by uninstalling then
reinstalling the RA cores on his SNES mini.

So we never broke it doin the Namco update which is all i wanted to know :)

**Comment by arcadez2003** [comment]
Some fella requested it, so i changed both Ghosts n Goblins and Ghouls N Ghosts over to 4-way
it wont make the ladders any easier i dont think which was his main reason for wanting the
switch.

You gotta have the knack for em ;)

**Comment by ghost** [comment]
it makes a big difference for me on 8 way if you go diagonal you player just stops that shouldnt happen at all. Im still crap at it but not as bad on a 4way lol

**Comment by ghost** [comment]
@KMFDManic  there is an issue with your having two branches not sure which one we are using.

https://github.com/KMFDManic/mame2003-xtreme.git has missing commits

https://github.com/KMFDManic-Cores/mame2003-xtreme.git not checked that one not sure what one we are using

**Comment by KMFDManic** [comment]
@grant2258 ignore the Core one. I did the pull request on this Repo, and didn't have any noticeable issues. Custom OSTs ran great for me, with sfii running even better.  If you feel any additional commits need added, they can go to this one.  I will be posting a Release tomorrow.  Thanks for looking out, mate! Keep me posted if u want anything else done:)

**Comment by ghost** [comment]
Well im having issues with it just see how users get on the commit history seems kinda messed up on your branch. Ive sent updates to you and gp start if you want then. Just let it go in the wild if other people get the same problems ill commit it.

The changes that fixed it for me are here just do a wget  https://github.com/grant2258/mame2003-xtreme/raw/ost_update/src/drivers/cps1.c if people are having issues big man

**Comment by KMFDManic** [comment]
@grant2258 so much appreciated! I will be releasing tomorrow with a nice shout out to you.

**Comment by ghost** [comment]
ok one more thing my good man there is a buildfix pull request you will need to pull. I have put a pull request in for gpstar repo we should all be synced up now makes it easier to work on things that way in the future.

**Comment by arcadez2003** [comment]
@KMFDManic If you want this supported on the MAME2003+ branch just create a pull sometime
and i'll make sure to merge it in.

On a seperate note someone added support for the Cyclone and Drz80 versions of the M68k
and Z80 CPU's, i dunno much about em TBH but i assume this is to boost the performance
with certain games on certain hardware.??

I dunno if it's of any use to the platforms that this core targets but i thought i'd mention it anyway.

**Comment by ghost** [comment]
ive already done s sf2 for plus needs some things added atm that ive added to xtreme. ill update it  soon here https://github.com/grant2258/mame2003-plus-barcade/commits/sf2_ost ill remove the dat update else it wont merge. Ill probably be better off redoing let me know if you want it done or you can do it yourself

**Comment by arcadez2003** [comment]
Aye i know you've got it done but you cant push to the main branch hence i tagged @KMFDManic 
as i assume he still can, but hey it's no bother @grant2258 if you wanna send it my way.......

https://github.com/arcadez2003/mame2003-plus-libretro

And i should hopefully be able to merge it into the main branch from there, no hurry for this BTW
i really only only posted up to let ya's know im interested in these nice sound sample additions
for Street Fighter and would like to see em supported in the main core.

Also that i wasn't blanking ya's as i was tagged and never responded as i was busy with other 
stuff is all

**Comment by arcadez2003** [comment]
I had to revert with regards to the SF samples as i got some build errors..........

https://github.com/libretro/mame2003-plus-libretro/commit/60df6b3b4b4121a6afffd27d41a12a8cb9e40144#comments

**Comment by KMFDManic** [comment]
@arcadez2003 I am fairly certain @grant2258 will be able to make the magic happen with Plus:)  

@gpstar81 Remember when we thought there was a limit to sample size?  The actual issue at hand is running out of memory on unload!  On PSC, which has exponentially more Ram to work with, the samples run fine, nearly no matter what the size.  On MD/SNES/NES Mini Classics, however, they run out of memory after 80 MB.  BUT, I will post a fix for this Next Update, which will simply use a page swap file on USB, 512 MB, as additional Ram, and the larger samples will work fine!  sfii, even at 112 MB won't work without the Swap File.  But, all good now:)  This also fixes a half dozen other things that limited Ram effects on the lesser spec Minis.  

I think an awesome choice for another Custom OST would be either Castlevania Vs with MESEN HD OST or Afterburner with FM Towns OST (which is exceptionally cool).  So many options!

Also, @grant2258 In your opinion, something we previously discussed...how difficult would it be to add Analog Controls and Options to Toggle them On/Off like with Plus, to Xtreme?  Just curious.  Plus has better overall compatibility on the Mini Classics, but Xtreme still maintains the slightly better performance/speed, which goes a huge way with games like Alien 3 and Jurassic Park.  This is where analog controls may really benefit Xtreme.

**Comment by gpstar81** [comment]
Yes I know about the sample size limitations is a memory problem. Mame is loading the entire sample pack for a rom into the system memory when it loads the rom by default. You can really notice this on the snes mini as it can take awhile for the game to initially load. Anyways, so systems with small amounts of ram like the snes mini obviously can not load 500mb worth of samples into the system memory, where as a PC normally would not have that problem.

That is when I did the compromise and modified the system to only preload small samples (less than 10mb) into memory (prevents game stuttering when the system reads and loads a sample, such as sound effect samples) while large samples (most likely music) get loaded when required and removed from memory when no longer used. So in theory, aside from really strange sample setups (a single 500mb sample file for example), large samples should mostly be ok for those small ram systems.

If you make each music sample larger than that 10mb limit I put in, the custom ost games should load much faster as the sample file will only be loaded on demand when requested vs loading everything at the initial loading of the rom which can take time on slow systems like the snes mini. There can be tradeoffs to this, for example with Street Fighter II, the sample music changes during the low health of a character, so there may be a loading pause/stutter if the fast music sequence is a large file that is not preloaded into memory, so using smaller sample sizes might be better for that example.

Refer to: /src/common.c and /src/common.h for more information.

**Comment by ghost** [comment]
@arcadez2003 i think this is more a makefile.common issue than anything else try changing all the .c in there to .o this is a symptom of why the video drivers and machine drivers needed renamed. Well we need .o and to clear duplicates out of the object files as well if there is any specially for static builds

@KMFDManic analog controls can be done yes. I remeber discussing hard drivin with you and how i was hesitant to add it. It can be done eaier for the user right now than it is in plus at  the moment. 

The complication in the plus port is RA wants to set controls is simply cant coupe with mapping wise and from its menus. If you forget about retropad and use the tab menu like normal mame078 you wont have any issues.  No more than you would in mame 078 anyway. That was the bain for me working in this port perfectly working stuff was ripped out like the osd menu to set volumes. The game information screen that tells you a game isint working. Ive restored some osd on the barcade version i make for myself to use. I will eventually return mame like it was in its native form just so much was ripped out and half working ra solutions where put into place maybe mark will finish them one day. I always maintained it should have been disabled by default until it was fixed no we have a cluster of complications for users. 

The biggest issue we have is mame by default expects an analog controller. Problems is hats and analog both map to u/d/l/r/ so for all games to work you have to choose a type of controller if you dont have analog you need to specify digital in input->general and map every analog control to digital hat and override the analog default thats set by mame for analog games

**Comment by KMFDManic** [comment]
@grant2258 If you are bored and wanna try this, I can try a direct compile of your fork, if you are able to get analog introduced.  It worked fine for me with Plus.  So, it might be viable for Xtreme, too.  Entirely up to you.  Not sure if it would be a fairly quick addition, or drawn out.  

@gpstar81 Thanks for the great response.  I am currently able to run Samples twice as much as our limitations, via Virtual Ram, which installs 512 MB Swap File to USB/SD.  I just recorded a video with it in action.  So, you will want to use this if you do run SNESC again in the future.  For PSC, No problems with any file size, really.  I was able to easily load 160+ MB Samples on MD/SNES/NES with the 512 MB Virtual Ram Page Swap File.  So, it will remain a permanent fixture in my set-up.  And, get this!

Double Dragon, Custom OST, MAME 2003 Xtreme, which we obviously had some slowdown issues, which previously were fixed up with some coding fixes grant2258 helped with..It runs better than ever on my current set-up, which is combination of Zram Crash Inhibitor, which allows additional ram to be used, directly working with the kernel; Xtreme Virtual Ram (512 MB Page Swap File); Xtreme Overclock (Yes, we can overclock to 1.344 Ghz now!, and easily toggle it on/off), in conjuction with Updated RetroArch Tweaks I implemented and the updated Core with speed hacks..Double Dragon Custom OST runs smooth as butter, and is ridiculously awesome.  I will link to it in action with these above outlined perimeters.  It also fixes ALL the other Custom OST games to run better, too.  So glad we got these on MAME 2003 Xtreme, rather than 2014!  2014 would still be running them slow as molasses!

Thanks again, guys!

**Comment by ghost** [comment]
@KMFDManic ill certainly get round to it at some point an we can discuss some things on implementation once its in place have a few other things in tinkering with atm.

**Comment by KMFDManic** [comment]
@grant2258 Don't worry about analog controls.  I have things working the way we need them to be for light gun games, racing games, etc, for now:)  

@gpstar81 I will be posting Virtual Ram HMOD for use with MD/SNES/NES.  It will put 512 MB swap file onto hard drive/flash drive.  Any file over 80 MB that can't unload properly, will work.  Tested it beyond 300 MB with MAME samples, just fine.  On PSC, you can do 300 MB+ fine, as long as you have enough power source.  Very happy with ALL the Custom OSTs!  Your technical explanation was quite helpful in understanding some of the load issues!  Hope you get a chance to try more of the recent stuff out.  I have put time into optimizing Amiga, too, as well as Amiga CD32.  A few others have been great with that, too, such as Sonninos and RSN8887.

@arcadez2003 I need to look into Elevator Action Returns for Xtreme, as someone reported that there is a part where you shoot lights or such, and it is supposed to get dark.  I need to actually play and see for myself, to know what they are talking about.  Not sure if any fixes were put into Plus for that, as they mentioned the dark effect works fine on Plus.  I need to also see if anything else cool should be added to Xtreme, now that I am pretty set on DOS, Amiga, etc!  Loving each and everything added to Plus!  

3:50 to see what works in Plus, but not Xtreme.

https://www.youtube.com/watch?v=e_DX939yGr4

**Comment by arcadez2003** [comment]
@KMFDManic You simply need to add these changes.........

https://github.com/libretro/mame2003-plus-libretro/commit/181f8a8b0ac0aafe75dda23dfe433b3f04a1411c

You also might be interested in the new c-chip emulation for Operation Wolf which i hooked up
in the plus core a wee while back, it adds extra enemy has found you levels and extra gameplay
features.

**Comment by KMFDManic** [comment]
@arcadez2003 not even sure how that commit slipped past me! Thanks so much:)

**Comment by arcadez2003** [comment]
@KMFDManic @grant2258 

Can one of ya's do me a favour.?? i just need the MAME2003.xml dat updated so all the romsets
added of late are supported inside it, i cant do it myself as i dont have the toolset i'll do the
other stuff like the catver ect etc.

The main reason i want this done is im finished on the core now and i dont wanna leave it
in such a state as some of the games are not supported in the docs.

i have fork here of the latest core............

https://github.com/arcadez2003/mame2003-plus-libretro

Regards

**Comment by ghost** [comment]
Ill do that for you matey not a problem just have to do some shopping and ill put the dat up for you.

**Comment by arcadez2003** [comment]
Thanks bigman, watch yoursel oot there wi aw that wind ;)

**Comment by ghost** [comment]
That should be you mate synced the weather is shite mate fairly windy and even more wet no doubt will wake up to some snow the morra

**Comment by arcadez2003** [comment]
Thanks for this cheers

**Comment by ghost** [comment]
nae bother at all big yean

**Comment by KMFDManic** [comment]
@grant2258 Thanks for stepping in on the .dat file  

@arcadez2003 I was prepping for another Release, since I am incorporating 3DO and a few other things into the mix.  Have you tried Smash TV on Plus and/or Xtreme?  Not entirely sure.  But, I will do a retest.  I think it may be bugged on both.  Xtreme, in my test, had a freeze glitch soon after starting.  Controls worked fine.  Plus, ran ok, but controls were goofy.  I will test further on both.  Otherwise, everything great!  Thanks again for the Elevator Action Returns fix:)

**Comment by KMFDManic** [comment]
@gpstar81 Posted latest Release, and SF II works near flawlessly in conjunction with Xtreme Virtual Ram and other things added.  In my test demonstration, the game had a little lag.  It is now running nice and smooth.  The math as far as what generally had issues unloading into memory on MD-SNES-NES, is roughly 80 MB (Arcade Wise).  This includes Core size and total size of game loaded.  So, on MAME 2010, for instance, attempting to run Metal Slug 5 would pretty much always crash, due to hitting over 80 MB for cumulative size.  This is no longer a problem with the 512 MB Swap File that installs on USB Flash/Hard Drive.  

Hopefully, you get a chance to try some stuff on the Minis.  Not sure if you ever got up to the newer hakchi version, as of yet, or are still on the older one.

@grant2258 @gpstar81 Now that this Release is out, and amazing feedback has been coming in...I have an idea for what I would like to try to incorporate next.  MANY games on MAME 2003 have bonked controls.  Typically, we'd go into MAME Menu, remap, etc.  This idea came up when I was talking to grant2258 about trying to get Hard Drivin' mapped!  But, what I would like to do, if possible, is just start remapping games, such as Ikari Warriors, Xybots, etc, where it would cleanly use L1 and R1 for rotary controls, etc.  And, either via coding via Core have them as presets.  Or, just have .cfgs override them, whichever works.  

And, taking this idea one step further...games that require nvram initialization to work first time, maybe do the nvrams, so they already exist, such as with NARC, Robotron, where without nvram, you can't start the game.  

Lastly, my other post about trying to get Gaiapolis to initialize.  I was able to get it finally working!  BUT, it primarily works with the 512 MB Swap File I mentioned above, since memory is super intensive on loading that specific game!  It is possible, if I test further, that once the nvram is generated, the Swap File is no longer needed, though.  I did a couple tests, and found it mainly had issues while trying to generate nvram.  Even Red Earth takes a super long time to generate nvram on PC, later MAMEs!

So, hopefully, the default preset controls for many of these stubborn games can be directly implemented into both MAME 2003 Plus/Xtreme.  And, using at least an SNES Controller as basis, as I don't want to remap L2 or R2, and maintain those for other stuff.

**Comment by ghost** [comment]
The defaults are esy to set. Your at an advantage because you have a snes controller you can map to that directly. Mame2003 and plus also need to support arcade panels ect. The control bit is very easy to do as  far as setting defaults go. A word to the wise though mame2003 cfg files will not be compatible with xtreme as the structures have changes you will probably get odd inputs if you mix 2003 and xtreme. It would be a wise move to let extreme use its own directory for cfg files.

**Comment by KMFDManic** [comment]
@grant2258 Since this topic is getting too long, gonna make a new one for the .cfg stuff.  I am gonna start with stuff such as Xybots!  I'd like to help you guys have these done for Plus, too!  They are pretty much unplayable without remapping.

**Comment by ghost** [comment]
Well that sounds like a plan. This will be more a tutorial on how to deal with it and how defaults arent always appropriate. It will become clear to you what the limitations are and the reasons why things are this way. 

The good news is this can be worked around in some cases drivers dont even use the defaults by the end of the discussion you will know more about inputs than you ever wanted too know and how to fix these the way you want it fixed.

**Comment by Wilstorm** [comment]
@KMFDManic - Hey, I was wondering if you had the source WAVs or FLACs for some of the custom OSTs? When looking at them they are pretty low quality rips. Since all the hard work of incorporating samples into the core is done I was thinking of trying to minimize the size but maintain a bit better quality. 

I spent some time looking at different sites with arcade sounds but I have no idea where the original source audio came from in these OSTs. If they are from a CD, OST, or whatnot. I could also open a new topic but didn't want to do that if you don't think it's worth checking into.

I did try the full stereo 16-bit moonwalk OST samples on a Pi 3 and it worked flawlessly. The load was fine, game speed & exiting. They are degraded samples also but they are still 325MB in size and worked great on the Pi 3 so I would be interested in bumping the quality if possible. I have no idea what limitation the Pi has but we can reduce them to 8-bit or 22KHz and keep some of the dynamic range that was lost I think.

**Comment by Wilstorm** [comment]
BTW, thumbs up on the Mandalorian! ;)

**Comment by gpstar81** [comment]
> @KMFDManic - Hey, I was wondering if you had the source WAVs or FLACs for some of the custom OSTs? When looking at them they are pretty low quality rips. Since all the hard work of incorporating samples into the core is done I was thinking of trying to minimize the size but maintain a bit better quality.
> 
> I spent some time looking at different sites with arcade sounds but I have no idea where the original source audio came from in these OSTs. If they are from a CD, OST, or whatnot. I could also open a new topic but didn't want to do that if you don't think it's worth checking into.
> 
> I did try the full stereo 16-bit moonwalk OST samples on a Pi 3 and it worked flawlessly. The load was fine, game speed & exiting. They are degraded samples also but they are still 325MB in size and worked great on the Pi 3 so I would be interested in bumping the quality if possible. I have no idea what limitation the Pi has but we can reduce them to 8-bit or 22KHz and keep some of the dynamic range that was lost I think.

@Wilstorm send me over your email and I will link you over to the source files that were used on the OST's.

**Comment by KMFDManic** [comment]
@gpstar81 I was actually going to send you an email today to see how things were for you in the current crazy world situation!  As of right now, I have been picking up extra hours due to coworkers having Covid.  I am also working my other job, which unfortunately had to close a location, also due to Covid making things more difficult for them to run more than one location.  So, they pretty much consolidated to one single location to make things more optimal.  I helped them entirely move, which took nearly 30 days, with how much was needed to be moved, done, etc!  

As far as Custom OSTs, I have been meaning to put the HIGHER quality ones up on my Google Drive, since the Virtual Ram Swap File method really helps load anything 80 MB or higher on MD/SNES/NES.  There was an issue with NBA Jam, where you couldn't play a full game with OST.  The game would just crash at a certain point.  But, it was due to the limitations of MD/SNES/NES.  Not sure which Mini/s you still have laying around.  But, I plan on putting another Release out 1-1-21, and it will have the necessary install hmods so people will have an easier time running the much larger OSTs.  If you wanna email me a link to the Larger Files, so I can audit to make sure I have correct ones, I will toss those up on Google Drive, then communicate with @Wilstorm as far as most optimal "smaller" ones, so people have choice of small or large.

I have been analyzing some of the Custom OST coding, and really am in awe of how nicely things execute, in regards to both you and bigbluefrontend.  I still have ideas for future projects, just nothing as elaborate as Marvel Vs Capcom, obviously.  Main ones that come to mind, which I think would really be nice...

1. Vs Castlevania with the Custom Mesen Emulator OST, which I have all the audio files for, already.  Mesen is far too slow for all of our platforms, and mainly good for PC usage.  But, man, those Custom NES OSTs are really damned awesome for that emulator! 18:15 in to see Castlevania with the Custom NES OST!

https://www.youtube.com/watch?v=IT9ZoJqLLqQ

Where I currently am at, is trying to get N64 running better, and succeeded on that front:

https://www.youtube.com/watch?v=8pt4NrMi5Y4

But, yeah, for now, I would love to try to audit and get the larger Custom OSTs onto Google Drive, so people can utilize them with this very Next Release.  HD Custom OSTs:)

**Comment by gpstar81** [comment]
@KMFDManic 

Check your email, I sent you over the link.

**Comment by Wilstorm** [comment]
@gpstar81 - I would really appreciate it. I just wanted to see what I could do with shrinking the samples but maintaining the best quality possibly. Keeping with the same format of splitting out the left/right channels and reducing them down but keeping them up around the 20kHz mark if possible. 

I can't seem to find your email on your profile page. I even followed you on Github here but I just don't see a link or email anywhere on your main page (https://github.com/gpstar81). If I nothing else I can post an email and remove it or at least use it to get you my main email. 

I was introduced to your work and OST's a long time ago through @KMFDManic over on the mame2003-plus page. A good guy and his video's are fantastic! I get inspired through his videos! 👍 The OST's really freshen up these classics and are a really nice touch.

@KMFDManic - I did watch you take down some AT-ATs the other day. Also the HD packs in the other video are great. They look sharp. My wife loves Mario Kart 64 and the N64 HD texture pack for it looks amazing and it runs perfect on the Pi 4. The Pi 3 struggles a bit. Her only interest in retro gaming is all things Nintendo but I'll take what I can get out of her as usually she just takes a pass on everything else.

**Comment by gpstar81** [comment]
@Wilstorm 

I updated my profile settings, you will find it on my profile now. Send me a email and i'll get back to you.

**Comment by Wilstorm** [comment]
@gpstar81 - Ok, got it and sent! :)

**Comment by KMFDManic** [comment]
@gpstar81 Thanks:)  With @Wilstorm audiophile knowledge, should be able to get some solid ones converted for those with less space.  These games are still very much a permanent fixture!  I still have other stuff I plan on updating for 2003 Xtreme.  Not sure if you noticed the nice addition for L1/R1 for Rotary games, such as Ikari Warriors.  And, L1/R1 for racing games, accelerate, braking.  Custom OSTs are definitely the top bill, though, and always will be!

Also, @Wilstorm With the N64 I am updating, the Kaze hacks, which are some of the best hacks around, for Mario 64, will be quite playable.

**Comment by Wilstorm** [comment]
@KMFDManic - Now you're taking my wife's language. She loves Mario, even on the older platforms like Super Mario World on SNES, Mario Bros. on NES, etc. She's actually better then me.

The idea of using the L1/R1 for rotary games like Ikari Warriors, Heavy Barrel, Guerilla War, Time Soldiers I think are the big ones. That seems like it would work very well, that is really clever!

**Comment by KMFDManic** [comment]
@Wilstorm I am doing an audit of all of the Custom OST source files, thanks to @gpstar81 I will throw them up on my Google Drive, and we can go over a few variables, as far as Audiophile and those who have less space.  There are commits for us on the Mini Classics, so we CAN run the larger OSTs on USB, too.  And, due to limited 256 MB Ram, unloads tend to crash on Arcade Games and files, at anything above 80 MB or so.  Using Virtual Ram Memory, via Swap File fixes this, and these files actually load without crashing!  It was definitely a little annoying and perplexing to be able to only play half a game of NBA Jam Custom OST, and debug realize it was running out of memory!

N64 Wise, I am happy with what I was able to exploit, as I now also have Vigilante 8 working decently, per my latest video.

@Grant2258 has mentioned Karnov is a tough game, so I am gonna do a funny silly video about Karnov and my attempts to beat it!

Lastly, another acquaintance, I had test Xtreme on Retropie 4, I will try to fix up the makefile so it can be compiled for you on Pi4, so you can test these additions out, too!  

All of you, stay safe!!!

**Comment by Wilstorm** [comment]
@KMFDManic - That sounds great, just let me know when your ready. You know the size target better than I do. That’s pretty funny about nbajam!

If any of the lower quality soundtrack files are available for purchase for a decent price I would snatch them up to improve the audio quality. Some are are pretty low around 128kbps MP3. I’m winding down the week early here to spend some time with family. 

Let me know when you post Karnov. I love that game, a true classic. Stay safe too! Have a great holiday and Merry Christmas to you good sir!! :)

**Comment by Wilstorm** [comment]
Just to clarify if there’s a soundtrack for purchase that would improve some of the OST’s just let me know and I would purchase it, to kind of make a contribution since you guys guys did all the work to put those beauties together!

**Comment by KMFDManic** [comment]
@Wilstorm I'll throw them up on my google drive today, and let you know once posted, so you can mess with them.  We can go from there.  I'll also do a fun Karnov video.

**Comment by KMFDManic** [comment]
@Wilstorm They are almost uploaded, and roughly 9 GB in size.  So, be prepared for a big download.  I will let you know once they are up.  Then, you can let me know once you have the link.  Here is a Karnov video, in the meantime!:)

https://www.youtube.com/watch?v=FWn7DLnO7Ck

**Comment by KMFDManic** [comment]
@Wilstorm here ya go, let me know once you've gotten the link...

https://drive.google.com/file/d/19DWDuMrIckpLVynHrvpGKR8_HC6sb33Y/view?usp=drivesdk

**Comment by Wilstorm** [comment]
@KMFDManic - Ok, all set here, a big thanks to you and @gpstar81, I really appreciate it!

Your Karnov video is spot on! I loved the opening comments (walking 20 miles/masks) and I agree It's a tough game for sure. Though it's been a long time since I've loaded it up and had a serious play session I've made it pretty far playing regular. The Karnov sign in Bad Dudes was classic!

The last game I really was deep into was Rastan. It is a lot of fun for me. That can be another tough game for sure though but I've always had a soft spot for it! :)

I hope you had a great Christmas and the New Year can't come to quickly enough if it'll better than the last! :)
